rfc/system/0001-flyte-execution-tags.md

```diff
@@ -0,0 +1,127 @@
+execution in Kubernetes. We'll create two new tables in the flyteadmin database.
+- ``execution_admin_tags`` is a join table, and it contains admin_tags ID and execution ID.
+- ``admin_tags`` saves all the tag names
+
+As a first step, we recommend that these tags are
+persisted associated with an execution and then ListExecutions API is updated
+to return all 
+ - filtered executions by tags with supported `and` `or` queries
+ - all associated executions with every execution
+ - limit total number of tags per execution to 10-15
+
+For the second step, we will support attach tags to the project, task, and workflow. In addition,
+we will enable users to update the tags of an execution after it has been created. This will be done
+through a flyteadmin API (need to create a new endpoint for updating the tags).
+
+Once this is implemented the UI and CLI can be updated to support these
+queries.
+
+### CLI Interface 
+
+A workflow or task can be executed using
+
+```bash
+pyflyte run --remote --tags '["hello", "world"]' test.py wf --input1=10
+```
+ (or equivalently in flytectl)
+
+flytectl and flyte remote can support filtering of executions by tags. Example
+in flytectl,
+```bash
+flytectl get execution -p flytesnacks -d development --filter.tags="hello,world"
+```
+
+### UI Interface
+All tags are treated the same way and allow search/filter and click based grouping.
+Users will get the regular executions view with all the tags available on each execution.
+The users are allowed to filter an execution simply by clicking on a tag and then all
+executions are filtered by that tag. 
+
+![Grouping / Filtering UX](https://raw.githubusercontent.com/flyteorg/static-resources/main/flyte/rfc/tags/labels-filter.png)
+
+
+### Support for descriptions
+
+Users want to describe their executions and especially once an experiment
+succeeds they may want to add a lot more description and data about the
+experiment. Thus, we propose to add descriptions to the execution as well.
+
+Allow add description when you start an execution
+```bash
+pyflyte run --remote --tags '["key1", "key2"]' --description "........" test.py wf --input1=10
+```
+ 
+It should be possible to add the description as a Markdown
+```bash
+pyflyte run --remote --tags '["key1", "key2"]' --description README.md test.py wf --input1=10
+```
+ 
+It should be possible to add a description for an execution in the UI, after
+the execution has been created.
+![UI Descriptions](https://raw.githubusercontent.com/flyteorg/static-resources/main/flyte/rfc/tags/description-edit.png)
+
+## 4 Metrics & Dashboards
+NA
+
+## 5 Drawbacks
+It is important to understand that this may add a little more stress to the
+metadata database. Back of the envelope calculation
+
+ -> 1 million executions * 20 tags each. 
+ -> Each tag is a plain string, which has 64 characters.
+ -> 1.28 * 10^9 bytes (assuming one byte per character" -> 1.28GB
+
+This is not significant and will increase as executions increase
```

Comment:
```suggestion
This is not significant even though it will increase as executions increase
```

Was this the intended meaning? Otherwise the sentence appears to be contradicting itself.